charon/tests/ui.rs

```diff
@@ -0,0 +1,177 @@
+//! Ui tests for the charon compiler. Each `<file>.rs` file in `ui/` will be passed to the charon
+//! driver. The corresponding pretty-printed llbc output will be stored in `<file>.llbc`, and CI
+//! will ensure these stay up-to-date.
+//!
+//! Files can start with special comments that affect the test behavior. Supported magic comments:
+//! see `HELP_STRING`.
+use anyhow::bail;
+use assert_cmd::prelude::CommandCargoExt;
+use indoc::indoc as unindent;
+use libtest_mimic::{Outcome, Test};
+use std::{
+    error::Error,
+    fs::read_to_string,
+    path::{Path, PathBuf},
+    process::Command,
+};
+use walkdir::{DirEntry, WalkDir};
+
+use charon_lib::cli_options::{CliOpts, CHARON_ARGS};
+
+enum TestKind {
+    PrettyLlbc,
+    KnownPanic,
+    Unspecified,
+}
+
+struct MagicComments {
+    test_kind: TestKind,
+}
```

Comment:
Why "magic"?

Reply:
I think this is what the ui-test crate calls them. I could call them something else if you have an idea